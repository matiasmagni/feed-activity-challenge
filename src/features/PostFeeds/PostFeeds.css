i.fas.fa-comments {
    color: #ffffff;
}
