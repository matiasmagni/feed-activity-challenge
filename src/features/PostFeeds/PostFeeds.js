import React, { useState, useEffect } from 'react';
import { Link } from "react-router-dom";
import axios from 'axios';
import config from '../../config';
import './PostFeeds.css';

const PostFeeds = () => {
    const [posts, setPosts] = useState([]);

    useEffect(() => {
        const fetchPosts = async () => {
            const result = await axios(config.API_URL.POSTS);
            setPosts(result.data);
        };

        fetchPosts();
    }, []);

    return (
        <div className="container-fluid video-player">
            <div className="container">
                <div className="row vr-gallery">
                    {posts.map(post => (
                        <div className="row" key={`post-${post.id}`}>
                            <div className="col-md-8 mb-4">
                                <div className="row">
                                    <div className="col-md-12 col-lg-7 pr-0 pd-md">
                                        <img src="images/gallery-img1.png" alt="" />
                                    </div>
                                    <div className="col-md-12 col-lg-5 light-bg cus-pd cus-arrow-left">
                                        <p><small>march 27, 2020</small></p>
                                        <h3>{post.title}</h3>
                                        <p>{post.body}</p>
                                    </div>
                                </div>
                            </div>
                            <div className="col-md-4 pl-4 mb-4">
                                <div className="card">
                                    <img className="card-img h-100" src="images/video-cover1.jpg" alt="" />
                                    <div className="card-img-overlay opacity text-center">
                                        <Link className="play-1" to={`/comments/${post.id}`}>
                                            <i className="fas fa-comments fa-2x"></i>
                                        </Link>
                                        <h5 className="card-title">View all comments...</h5>
                                    </div>
                                </div>
                            </div>
                        </div>
                    ))}
                    <div className="col-md-12 text-center">
                        <a href="/#" className="btn">LOAD MORE</a>
                    </div>
                </div>
            </div>
        </div>
    );
}

export default PostFeeds;