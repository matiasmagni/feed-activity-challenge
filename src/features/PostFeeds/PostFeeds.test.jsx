import React from 'react';
import axios from 'axios';
import MockAdapter from 'axios-mock-adapter';
import { cleanup } from '@testing-library/react';
import { act as domAct } from 'react-dom/test-utils';
import { Provider } from 'react-redux';
import { act as testAct } from 'react-test-renderer';
import store from '../../redux/store';
import PostFeeds from './PostFeeds';
import config from '../../config';
import renderWithRouter from '../../__mocks__/renderWithRouter';

const mockGetPosts = new MockAdapter(axios);

const postApiData = [
  {
    userId: 1,
    id: 1,
    title: 'sunt aut facere repellat provident occaecati excepturi optio reprehenderit',
    body: 'quia et suscipit\nsuscipit recusandae consequuntur expedita et cum\nreprehenderit molestiae ut ut quas totam\nnostrum rerum est autem sunt rem eveniet architecto',
  },
  {
    userId: 1,
    id: 2,
    title: 'qui est esse',
    body: 'est rerum tempore vitae\nsequi sint nihil reprehenderit dolor beatae ea dolores neque\nfugiat blanditiis voluptate porro vel nihil molestiae ut reiciendis\nqui aperiam non debitis possimus qui neque nisi nulla',
  },
];

afterEach(cleanup);

describe('PostFeeds component tests.', () => {
  it('PostFeeds component renders correctly.', () => {
    let postFeeds = null;

    domAct(() => {
      testAct(() => {
        postFeeds = renderWithRouter(<Provider store={store}><PostFeeds /></Provider>);
        mockGetPosts.onGet(config.API_URL.POSTS).reply(200, postApiData);
      });
    });

    const { asFragment } = postFeeds;
    expect(asFragment()).toMatchSnapshot();
  });

  it('PostFeeds render all posts from API correctly.', () => {
    let postFeedsComponent = null;

    testAct(() => {
      postFeedsComponent = renderWithRouter(<Provider store={store}><PostFeeds /></Provider>);
      mockGetPosts.onGet(config.API_URL.POSTS).reply(200, postApiData);
    });

    const { findByText } = postFeedsComponent;

    postApiData.forEach((data) => {
      ['title', 'body'].forEach(async (field) => {
        expect(
          await findByText((content) => content.includes(data[field].substring(0, 15))),
        ).toBeInTheDocument();
      });
    });
  });
});
