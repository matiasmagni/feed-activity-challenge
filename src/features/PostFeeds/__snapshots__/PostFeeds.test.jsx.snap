// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`PostFeeds component tests. PostFeeds component renders correctly. 1`] = `
<DocumentFragment>
  <div
    class="container-fluid video-player"
  >
    <div
      class="container"
    >
      <div
        class="row vr-gallery"
      />
    </div>
  </div>
</DocumentFragment>
`;
